sycl/doc/extensions/experimental/sycl_ext_oneapi_complex.asciidoc

```diff
@@ -0,0 +1,384 @@
+            friend constexpr bool operator==(value_type x, const complex<value_type> &w);
+
+            /// Compares complex numbers z and w and returns true if they are different, otherwise false.
+            friend constexpr bool operator!=(const complex<value_type> &z, const complex<value_type> &w);
+            ///Compares complex number z and real y and returns true if they are different, otherwise false.
+            friend constexpr bool operator!=(const complex<value_type> &z, value_type y);
+            /// Compares real x and complex number w and returns true if they are different, otherwise false.
+            friend constexpr bool operator!=(value_type x, const complex<value_type> &w);
+
+            /// Reads a complex number from is.
+            /// Not allowed in device code.
+            template <class C, class T> friend std::basic_istream<C, T> &operator>>(std::basic_istream<C, T> &is, complex<value_type> &);
+            /// Writes to os the complex number z in the form (real,imaginary).
+            /// Not allowed in device code.
+            template <class C, class T> friend std::basic_ostream<C, T> &operator<<(std::basic_ostream<C, T> &os, const complex<value_type> &);
+            /// Streams the complex number z in the format "(real,imaginary)" into `sycl::stream` x and return the result.
+            friend const sycl::stream &operator<<(const sycl::stream &x, const complex<value_type> &z);
+
+} // namespace experimental
+} // namespace oneapi
+} // namespace ext
+} // namespace sycl
+```
+
+=== Mathematical operations
+
+This proposal adds to the `sycl::ext::oneapi::experimental` namespace, math
+functions accepting the complex types `complex<sycl::half>`, `complex<float>`,
+`complex<double>` as well as the scalar types `sycl::half`, `float` and `double`
+for the SYCL math functions, `abs`, `acos`, `asin`, `atan`, `acosh`, `asinh`,
+`atanh`, `arg`, `conj`, `cos`, `cosh`, `exp`, `log`, `log10`, `norm`, `polar`,
+`pow`, `proj`, `sin`, `sinh`, `sqrt`, `tan`, and `tanh`.
+
+These functions are available in both host and device code, and each math
+function should follow the C++ standard for handling NaN's and Inf values.
+
+Note: In the case of the `pow` function, additional overloads have been added
+to ensure that for their first argument `base` and second `argument`:
+
+If `base` and/or `exponent` has type `complex<sycl::half>` or `sycl::half`,
+then `pow(base, exponent)` has the same effect as
+`pow(complex<sycl::half>(base), complex<sycl::half>(exponent))`.
+
+Otherwise, if `base` and/or `exponent` has type `complex<float>`, `float`,
+then `pow(base, exponent)` has the same effect as
+`pow(complex<float>(base), complex<float>(exponent))`.
+
+Otherwise, if `base` and/or `exponent` has type `complex<double>`, `double`,
+then `pow(base, exponent)` has the same effect as
+`pow(complex<double>(base), complex<double>(exponent))`.
```

Comment:
These need to be ordered from highest precision to lowest (i.e. `double`, then `float`, then `half`).  As you have the wording now, `pow(complext<half>, complex<double>)` would be evaluated as `pow(complext<half>, complex<half>)`, which is certainly not what you want.

I think it would also be more readable if these three paragraphs were a bulleted list.

Reply:
I've updated this section with my latest commit, which should resolve this comment and the next one.